File: HelloWorld/ObjectOriented.java
// Compile command: javac -encoding "UTF-8" ObjectOriented.java
// Execute commands:
//   java -Dfile.encoding="UTF-8" ObjectOriented
//   java -Dfile.encoding="UTF-8" ObjectOriented "José Carlos Monteiro"
//   java -Dfile.encoding="UTF-8" ObjectOriented Universe

public class ObjectOriented {

  private final String name;

  public ObjectOriented() {
    name = "World";
  }

  public ObjectOriented( final String name ) {
    this.name = name;
  }

  public String say() {
    return String.format( "Hello, %s!", name );
  }

  public static void main( final String[] arguments ) {
    final ObjectOriented greeter = ( arguments.length == 0 ) ? new ObjectOriented() : new ObjectOriented( arguments[ 0 ] );
    System.out.println( greeter.say() );

    System.out.println( "» José Carlos Monteiro «" );
  }

}

File: HelloWorld/ObjectOriented.cs
// Compile command: mcs ObjectOriented.cs
// Execute commands:
//   ObjectOriented
//   ObjectOriented "José Carlos Monteiro"
//   ObjectOriented Universe

using System;

public class ObjectOriented
{

  #region Fields.

  private string name;

  #endregion

  #region Constructors.

  public ObjectOriented()
  {
    name = "World";
  }

  public ObjectOriented( String name )
  {
    this.name = name;
  }

  #endregion

  #region Public Methods.

  public string say()
  {
    return String.Format( "Hello, {0}!", name );
  }

  public static void Main( string[] arguments )
  {
    ObjectOriented greeter = ( arguments.Length == 0 ) ? new ObjectOriented() : new ObjectOriented( arguments[ 0 ] );
    Console.WriteLine( greeter.say() );

    Console.WriteLine( "» José Carlos Monteiro «" );
  }

  #endregion

}